projects/map-ol/src/lib/map-ol.component.ts

```diff
@@ -170,6 +170,12 @@ export class MapOlComponent implements OnInit, AfterViewInit, AfterViewChecked,
           if (ollayer) {
             if (ollayer.getZIndex() !== layers.indexOf(layer) + otherlayerslength) {
               ollayer.setZIndex(layers.indexOf(layer) + otherlayerslength);
+              // addresses an issue in openlayers: https://github.com/openlayers/openlayers/issues/6654
+              if (ollayer instanceof olLayerGroup) {
+                (ollayer as olLayerGroup).getLayersArray().map(l => {
```

Comment:
Here I would use `(ollayer as olLayerGroup).getLayers()` so you work with the a `olCollection` https://openlayers.org/en/latest/apidoc/module-ol_layer_Group-LayerGroup.html#getLayers also `getLayersArray` is not listed anymore in there API https://openlayers.org/en/latest/apidoc/module-ol_layer_Group-LayerGroup.html.